src/preprocess/visitor.ts

```diff
@@ -397,17 +398,34 @@ export class Visitor {
   visit$split(node: yaml.$split, path: string, env: Env): string[] {
     if (_.isArray(node.data) && node.data.length === 2) {
       const [delimiter, str]: [string, string] = node.data;
-      const escapedDelimiter = escapeStringRegexp(delimiter);
-      return this.visitNode(str, path, env)
-        .toString()
-        .replace(new RegExp(`${escapedDelimiter}+$`), '') // Remove trailing delimiters
-        .split(delimiter);
+      const visitedDelimiter = this.visitNode(delimiter, path, env);
+      return _.dropRightWhile(
+        this.visitNode(str, path, env).toString().split(visitedDelimiter),
+        (s) => s === "" // Remove trailing delimiters
```

Comment:
let's update the comment here to reflect that we're not removing the trailing delimiters anymore but are now removing the garbage they can create.